compiler/optimizer/LoopVersioner.cpp

```diff
@@ -4295,8 +4302,13 @@ void TR_LoopVersioner::versionNaturalLoop(TR_RegionStructure *whileLoop, List<TR
       for (; search.hasTreeTop(); search.advance())
          {
          TR::TreeTop *tt = search.currentTreeTop();
+         if (removedNodes.contains(tt->getNode()))
+            continue;
+
```

Comment:
Is this safe in all cases ? Is it possible that we still need to account for the case (even if it's due to performTransformation disallowing a transformation, e.g. removal of some node) when the nodes in the list are not actually removed ?

Reply:
The perform transformations are around the points where the nodes are added to the list. There is no perform once added to the list because we make decisions based on what we plan to version. The list we check is based on what we know we are going to do not what we might do so this should be safe and matches the versioning of HCR guards and other similar operations elsewhere